### ISSUE TITLE
Cannot find L1 loss in training code

### ISSUE THREAD

**Issue by zhui1** [Eric-mingjie/network-slimming#19]

**Comment by Eric-mingjie** [comment]
https://github.com/Eric-mingjie/network-slimming/blob/master/main.py#L126

**Comment by zhui1** [comment]
the loss we find in main.py is 
`def updateBN():
    for m in model.modules():
        if isinstance(m, nn.BatchNorm2d):
            m.weight.grad.data.add_(args.s*torch.sign(m.weight.data))  # L1` 
why this is not what is said in the paper as:
> In our experiment, we choose g(s) = |s|, which is known as L1-norm and widely used to achieve sparsity.

**Comment by Eric-mingjie** [comment]
The derivative of abs() function is the sign() function.

**Comment by PeiqinSun** [comment]
为什么不直接在loss定义正则化项, 而是直接写在梯度反传上呢? 这样做两者是等价还是在某些情况下有区别啊? 比如采用不同的optimizer?
